packages/material-ui-utils/src/requirePropFactory.js

```diff
@@ -12,6 +17,17 @@ export default function requirePropFactory(componentNameInError) {
   ) => {
     const propFullNameSafe = propFullName || propName;
 
+    const defaultTypeChecker = prevPropTypes?.[propFullNameSafe];
+
+    if (defaultTypeChecker) {
+      PropTypes.checkPropTypes(
+        { [propName]: defaultTypeChecker },
+        props,
+        location,
+        componentNameInError,
+      );
+    }
```

Comment:
Why calling `checkPropTypes` when React already does it?

```diff
diff --git a/packages/material-ui-utils/src/requirePropFactory.js b/packages/material-ui-utils/src/requirePropFactory.js
index 9955b4176b..13d19d7a29 100644
--- a/packages/material-ui-utils/src/requirePropFactory.js
+++ b/packages/material-ui-utils/src/requirePropFactory.js
@@ -14,18 +14,14 @@ export default function requirePropFactory(componentNameInError, Component) {
     componentName,
     location,
     propFullName,
+    ...args
   ) => {
     const propFullNameSafe = propFullName || propName;

     const defaultTypeChecker = prevPropTypes?.[propFullNameSafe];

     if (defaultTypeChecker) {
-      PropTypes.checkPropTypes(
-        { [propName]: defaultTypeChecker },
-        props,
-        location,
-        componentNameInError,
-      );
+      defaultTypeChecker(props, propName, componentName, location, propFullName, ...args);
     }

     if (typeof props[propName] !== 'undefined' && !props[requiredProp]) {
diff --git a/packages/material-ui-utils/src/requirePropFactory.test.js b/packages/material-ui-utils/src/requirePropFactory.test.js
index 7c72e3bbda..3ae483b982 100644
--- a/packages/material-ui-utils/src/requirePropFactory.test.js
+++ b/packages/material-ui-utils/src/requirePropFactory.test.js
@@ -161,7 +161,6 @@ describe('requirePropFactory', () => {
             'Test',
           );
         }).toErrorDev([
-          'Warning: Failed prop type: Invalid prop `test` of type `boolean` supplied to `Test`, expected `string`.',
           'Warning: Failed prop type: The prop `test` of `Test` can only be used together with the `otherProp` prop.',
         ]);
       });
```

Reply:
> Do we still benefit from deduplication?

No, we don't:

<img width="593" alt="Capture d’écran 2021-01-15 à 13 06 36" src="https://user-images.githubusercontent.com/3165635/104725555-8a960180-5732-11eb-925e-e3ffb83ad028.png">

Reply:
You are correct, we don't need `checkPropTypes` here it was my experiment. In the test however, we should test the removed line, not the other :) Fixed with https://github.com/mui-org/material-ui/pull/24423/commits/abad297cc1b16ec6d5ed24c579d5e48d1197bcf4